packages/perf/test/index.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { stubInterface } from 'sinon-ts'
+import { defaultInit, perfService } from '../src/index.js'
+import type { TransportManager } from '@libp2p/interface-internal/transport-manager'
+
+export async function createComponents (): Promise<Components> {
+  const peerId = await createEd25519PeerId()
+
+  const events = new EventEmitter()
+
+  const components = defaultComponents({
+    peerId,
+    registrar: mockRegistrar(),
+    upgrader: mockUpgrader(),
+    datastore: new MemoryDatastore(),
+    transportManager: stubInterface<TransportManager>(),
+    connectionGater: mockConnectionGater(),
+    events
+  })
+
+  components.peerStore = new PersistentPeerStore(components)
+  components.connectionManager = new DefaultConnectionManager(components, {
+    minConnections: 50,
+    maxConnections: 1000,
+    autoDialInterval: 1000,
+    inboundUpgradeTimeout: 1000
+  })
+
+  return components
+}
+
+describe('perf', () => {
+  let localComponents: Components
+  let remoteComponents: Components
+
+  beforeEach(async () => {
+    localComponents = await createComponents()
+    remoteComponents = await createComponents()
+
+    await Promise.all([
+      start(localComponents),
+      start(remoteComponents)
+    ])
+  })
+
+  afterEach(async () => {
+    await Promise.all([
+      stop(localComponents),
+      stop(remoteComponents)
+    ])
+  })
+
+  it('should run perf', async () => {
+    const client = perfService(defaultInit)(localComponents)
+    const server = perfService(defaultInit)(remoteComponents)
+
+    await start(client)
+    await start(server)
+
+    // simulate connection between nodes
+    const [localToRemote, remoteToLocal] = connectionPair(localComponents, remoteComponents)
+    localComponents.events.safeDispatchEvent('connection:open', { detail: localToRemote })
+    remoteComponents.events.safeDispatchEvent('connection:open', { detail: remoteToLocal })
+
+    // Run Perf
+    await expect(client.perf(localToRemote, 1024n, 1024n)).to.eventually.be.fulfilled()
+  })
+
+  it('should output bandwidth', async () => {
+    const client = perfService(defaultInit)(localComponents)
+    const server = perfService(defaultInit)(remoteComponents)
+
+    await start(client)
+    await start(server)
+
+    // simulate connection between nodes
+    const [localToRemote, remoteToLocal] = connectionPair(localComponents, remoteComponents)
+    localComponents.events.safeDispatchEvent('connection:open', { detail: localToRemote })
+    remoteComponents.events.safeDispatchEvent('connection:open', { detail: remoteToLocal })
+
+    let downloadBandwidth = 0
+    let uploadBandwidth = 0
+
+    for (let i = 0; i < 5; i++) {
+      // Run Perf
+      downloadBandwidth += await client.measureDownloadBandwidth(localToRemote, 10485760n)
+      uploadBandwidth += await client.measureUploadBandwidth(localToRemote, 10485760n)
+    }
+
+    // eslint-disable-next-line no-console
+    console.log('Upload bandwidth:', Math.floor(downloadBandwidth / 5), 'B/s')
```

Comment:
Why are you dividing by 5?

Reply:
The is an average bandwidth, so run 5 iterations, sum them, then divide by 5.

Reply:
If this is just a unit test, can we do a single run and smaller value. Something bigger than a packet size. 8KiB is probably fine.